internal/pkg/server/agent.go

```diff
@@ -115,6 +115,42 @@ func (a *Agent) Run(ctx context.Context) error {
 		}
 		return p
 	})
+	a.agent.RegisterDiagnosticHook("fleet-server api tls diag", "fleet-server's API TLS config", "fleet-server-api-tls.txt", "text/plain", func() []byte {
+		if a.srv == nil {
+			log.Warn().Msg("Diagnostics hook failure fleet-server is nil.")
+			return []byte(`Diagnostics hook failure fleet-server is nil`)
+		}
+		cfg := a.srv.GetConfig()
+		if cfg == nil || len(cfg.Inputs) == 0 {
+			log.Warn().Msg("Diagnostics hook failure config is nil.")
+			return []byte(`Diagnostics hook failure config is nil`)
+		}
+		return cfg.Inputs[0].Server.TLS.DiagCerts()()
+	})
+	a.agent.RegisterDiagnosticHook("fleet-server output tls diag", "fleet-server's output TLS config", "fleet-server-output-tls.txt", "text/plain", func() []byte {
+		if a.srv == nil {
+			log.Warn().Msg("Diagnostics hook failure fleet-server is nil.")
+			return []byte(`Diagnostics hook failure fleet-server is nil`)
+		}
+		cfg := a.srv.GetConfig()
+		if cfg == nil {
+			log.Warn().Msg("Diagnostics hook failure config is nil.")
+			return []byte(`Diagnostics hook failure config is nil`)
+		}
+		return cfg.Output.Elasticsearch.TLS.DiagCerts()()
+	})
+	a.agent.RegisterOptionalDiagnosticHook("CONN", "fleet-server output request diag", "fleet-server output request trace diagnostics", "fleet-server-output-request.txt", "text/plain", func() []byte {
+		if a.srv == nil {
+			log.Warn().Msg("Diagnostics hook failure fleet-server is nil.")
+			return []byte(`Diagnostics hook failure fleet-server is nil`)
+		}
+		cfg := a.srv.GetConfig()
+		if cfg == nil {
+			log.Warn().Msg("Diagnostics hook failure config is nil.")
+			return []byte(`Diagnostics hook failure config is nil`)
+		}
+		return cfg.Output.Elasticsearch.DiagRequests(ctx)
```

Comment:
Where is this `ctx` coming from? Doesn't seem to be inside of this hook call. If the context is outside of the request hook that seems like that could be incorrect.

Reply:
This context is the "root" context for fleet-server running agent.
It can be cancelled when the fleet-server is shutdown (by signal); the request-hook will abort in this case

Reply:
This should add a timeout to it then, so use that context but add a timeout. We should ensure that there is atleast a maximum amount of time that the `DiagRequests` will take.